framework/src/modules/chain/components/storage/entities/temp_block.js

```diff
@@ -193,6 +195,33 @@ class TempBlock extends BaseEntity {
 			)
 			.then(result => result);
 	}
+
+	/**
+	 * Delete all rows in table
+	 *
+	 * @param {Object} [tx]
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	truncate(tx = null) {
+		return this.adapter.executeFile(
+			this.SQLs.truncate,
+			{},
+			{ expectedResultCount: 0 },
+			tx,
+		);
+	}
+
+	/**
+	 * Check if table is empty (true)
+	 *
+	 * @param {Object} [tx]
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	isEmpty(tx = null) {
+		return this.adapter
+			.executeFile(this.SQLs.isEmpty, {}, {}, tx)
+			.then(([result]) => (result ? result.bool : true));
```

Comment:
In case there are empty rows returned from the database, the result will be undefined which is falsy so the function will return true. Also, in the case when query returns non-empty rows, the value of result will be false hence the function will still return true. Hence, I am unsure when `([result]) => (result ? result.bool : true)` will return `result.bool`, and what actually is `result.bool`?
The tests seem to be correct, but  I am stiill confused by how it works :D

Reply:
I'm not sure what you mean with empty rows here?

When the table contains rows, it will return `false`. 
When there are no rows, no result is returned (that's why I removed `expectedResultCount` as it can be 0 or 1). 

So, if it contains rows -> return `false`.
No rows, result is undefined -> return `true`.

The `result.bool` comes from pgpromise as I'm only returning a bool as a result, it appends it under the `bool` property.

Reply:
By empty, I meant no rows. I was confused how value of the query result is assigned to result.bool property, but now I understand. Thanks for the explanation!